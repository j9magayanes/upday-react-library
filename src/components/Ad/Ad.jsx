
import React from 'react';
import { Image } from '../Image/Image'
import  'upday-web-components/components/adWrapper'
import './ad.css';

export const Ad = ({ src, label, args, }) => {
    return (
        <figure>
          <ad-wrapper class="ad-block grow">
              <h3>Story continues below advertisement</h3>
              <image-component args={args} src={src}/>
          </ad-wrapper>
      </figure>
    )
}