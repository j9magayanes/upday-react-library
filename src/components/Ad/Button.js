import Vue from 'vue';
import vueCustomElement from 'vue-custom-element';
import Button from './Button.vue';

Vue.use(vueCustomElement);

// Register the Vue component as a custom element
Vue.customElement('vue-counter', Button);
