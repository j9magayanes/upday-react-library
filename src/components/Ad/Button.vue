
<template>
  <div>
   <h1>Button</h1>
  </div>
</template>

<script>
export default {
  // eslint-disable-next-line vue/multi-word-component-names
  name: 'Button',
  props: {

  }
}
</script>