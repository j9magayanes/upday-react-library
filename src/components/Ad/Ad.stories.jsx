import React from 'react';

import { Ad } from './Ad';


export default {
  title: 'Form/Ad',
  component: Ad,
};

const Template = () =><Ad />;


export const Default = Template.bind({});