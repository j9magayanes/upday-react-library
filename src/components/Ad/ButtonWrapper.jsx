import React, { useRef, useEffect } from 'react';
import Vue from 'vue';
import Button from './Button.vue'; // Import your Vue component

const ButtonWrapper = () => {
  const vueContainerRef = useRef(null);

  useEffect(() => {
    const vm = new Vue({
      el: vueContainerRef.current,
      render: (h) => h(Button),
    });

    return () => {
      vm.$destroy();
    };
  }, []);

  return <div ref={vueContainerRef} />;
};

export default ButtonWrapper;
