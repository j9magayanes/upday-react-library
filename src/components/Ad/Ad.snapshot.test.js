import React from 'react';
import renderer from 'react-test-renderer';
import { Ad } from './Ad';

test('renders Ad component correctly', () => {
  const props = {
    src: 'your-image-src',
    label: 'your-label',
    args: { /* your args object */ },
  };

  const component = renderer.create(<Ad {...props} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
