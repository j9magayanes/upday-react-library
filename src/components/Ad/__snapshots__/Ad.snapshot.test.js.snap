// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Ad component correctly 1`] = `
<figure>
  <ad-wrapper
    class="ad-block grow"
  >
    <h3>
      Story continues below advertisement
    </h3>
    <image-component
      args={Object {}}
      src="your-image-src"
    />
  </ad-wrapper>
</figure>
`;
